asapdiscovery-docking/asapdiscovery/docking/modeling.py

```diff
@@ -623,3 +632,141 @@ def remove_extra_ligands(mol, lig_chain=None):
         if oechem.OEAtomGetResidue(a).GetExtChainID() != lig_chain:
             mol_copy.DeleteAtom(a)
     return mol_copy
+
+
+def check_completed(d, prefix):
+    """
+    Check if this prep process has already been run successfully in the given
+    directory.
+
+    Parameters
+    ----------
+    d : str
+        Directory to check.
+
+    Returns
+    -------
+    bool
+        True if both files exist and can be loaded, otherwise False.
+    """
+
+    if (not os.path.isfile(os.path.join(d, f"{prefix}_prepped_receptor_0.oedu"))) or (
+        not os.path.isfile(os.path.join(d, f"{prefix}_prepped_receptor_0.pdb"))
+    ):
+        return False
+
+    try:
+        du = oechem.OEDesignUnit()
+        oechem.OEReadDesignUnit(
+            os.path.join(d, f"{prefix}_prepped_receptor_0.oedu"), du
+        )
+    except Exception:
```

Comment:
it's a bit crude to catch exceptions like this but I'm guessing it's because OE doesn't raise exceptions in a more catch-able way. How about doing an `os.path.exists()` and catching a `FileNotFoundError` before the checks with OE? That would make the issue a lot clearer to the user.

Reply:
I agree, we want to avoid catching these generic exceptions, what we could do here if we don't know the actual exception is at least to raise a warning and let the users know how it failed. 

We can raise a warning such as `warning(e)` where `e` whould be the exception, as in `Exception as e`.

Reply:
This should be improved with this issue #185